internal/server/errors.go

```diff
@@ -78,20 +72,11 @@ func sendAPIError(c *gin.Context, err error) {
 	case errors.Is(err, context.DeadlineExceeded):
 		resp.Code = http.StatusGatewayTimeout // not ideal, but StatusRequestTimeout isn't intended for this.
 		resp.Message = "request timed out"
-
-	default:
-		log = logging.L.WithOptions(zap.AddCallerSkip(1)).Error
 	}
 
-	log("api request error",
```

Comment:
This log is extremely similar to the log line produced by the logging middleware with the addition of the error. Instead of producing the same line twice, attach the error to the context and consume it in the middleware

Reply:
I like this idea, I was thinking of a similar thing here: https://github.com/infrahq/infra/pull/2400#discussion_r908779508

But as Bruce said below there are other implications for this change.  

I think we should hold off on that for now. Switching logging libraries is already a huge change, we should probably try to limit any other changes in this PR.

Reply:
Removing for now